src/main/java/ai/deepcode/javaclient/core/Base64EncodeRequestInterceptor.java

```diff
@@ -0,0 +1,63 @@
+package ai.deepcode.javaclient.core;
+
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okio.Buffer;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.zip.GZIPOutputStream;
+
+public class Base64EncodeRequestInterceptor implements Interceptor {
+  private static final String CONTENT_ENCODING_HEADER = "Content-Encoding";
+  private static final String MEDIA_TYPE_OCTET_STREAM = "application/octet-stream";
+  private static final String MEDIA_TYPE_OCTET_STREAM_GZIP = "gzip";
+
+  @SuppressWarnings("NullableProblems") // we don't have @NotNull annotation
```

Comment:
we have and use @NotNull annotation in the project

Reply:
Somehow we thought with @bastiandoetsch that `@NotNull` from `org.jetbrains.annotations.*` is not the right one. I see now that we use it in other code parts, added it back.